test/org/zaproxy/zap/testutils/websocket/server/NanoWebSocketTestServer.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.testutils.websocket.server;
+
+import fi.iki.elonen.NanoWSD;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Creates a simple WebSocket Test Server. Stores every connection was established
+ */
+public class NanoWebSocketTestServer extends NanoWSD {
+   
+    private List<NanoWebSocketConnection> connectionsList;
+    private final static String DEFAULT_PING_MESSAGE = "PING";
+    
+    public NanoWebSocketTestServer(int port) {
+        super(port);
```

Comment:
Could call `this(null, port);` instead, which would avoid initialise the `connectionsList` in this constructor.